### PR TITLE
picture glance: better handle if no title

### PR DESCRIPTION
**Pull request by balloob**
While experimenting for https://github.com/home-assistant/ui-schema/issues/104#issuecomment-406161508, I realized that we don't handle the optional title well.

Before:
screen shot 2018-07-19 at 12 46 18 pm

After:
screen shot 2018-07-19 at 12 47 03 pm

### CODE REVIEW

**Comment by c727** [comment]
We should also put both arrays in dom-if in case of title and dialog entities only

File: src/panels/lovelace/cards/hui-picture-glance-card.js [hunk 1]
```diff
@@ -83,29 +83,27 @@ class HuiPictureGlanceCard extends NavigateMixin(LocalizeMixin(EventsMixin(Polym
...
-            <template is="dom-repeat" items="[[_entitiesToggle]]">
-              <template is="dom-if" if="[[_showEntity(item.entity, hass.states)]]">
-                <paper-icon-button
-                  on-click="_callService"
-                  class$="[[_computeButtonClass(item.entity, hass.states)]]"
-                  icon="[[_computeIcon(item.entity, hass.states)]]"
-                  title="[[_computeTooltip(item.entity, hass.states)]]"
-                ></paper-icon-button>
-              </template>
+            <template is="dom-repeat" items="[[_computeVisible(_entitiesToggle, hass.states)]]">
```
Comment (c727) [inline src/panels/lovelace/cards/hui-picture-glance-card.js]: Does this also move dialog[] to the right side if toggle[] is empty?

**Comment by balloob** [comment]
Done. 

Only downside is that if there are only toggle entities, they are shown on the left:

image

**Comment by balloob** [comment]
So the trick is in the `justify-content: space-between`. If we render a div that contains the dialog or toggle entities, it will make sure it is part of the calculation. If we conditionally not render the div, they are taken out and space is given to other components.

Permutations with the divs hidden if no entities:

image

**Comment by c727** [comment]
`justify-self: flex-end;`?

File: src/panels/lovelace/cards/hui-picture-glance-card.js [hunk 2]
```diff
@@ -150,6 +160,10 @@ class HuiPictureGlanceCard extends NavigateMixin(LocalizeMixin(EventsMixin(Polym
     });
   }
 
+  _computeVisible(collection, states) {
+    return collection.filter(el => el.entity in states);
+  }
+
   _showEntity(entityId, states) {
```
Comment (c727) [inline src/panels/lovelace/cards/hui-picture-glance-card.js]: can be removed now

**Comment by balloob** [comment]
Check the netlify deploy in the commit status to see impact of this PR 👍

**Comment by balloob** [comment]
That would be nice but since at build time we only have access to 1 version, that won't work. You can however always open a browser and open https://home-assistant-lovelace-gallery.netlify.com/ (the master build). We should add some links to open the master from each demo page.

**Comment by jeradM** [comment]
You could conditionally set `margin-left: auto` on the toggles div to push it right if there are no dialog entities

**Comment by balloob** [comment]
Rebased on latest master to include more demos for picture-glance.

**Comment by balloob** [comment]
@jeradM margin-left: auto doesn't work if there are 3 elements:

image

However, it is all fixed if I always render both divs containing the entities 👍 see netflix preview.

**Pull request merged**